ulib/libax/src/cbindings/socket.rs

```diff
@@ -400,23 +444,128 @@ pub unsafe extern "C" fn ax_shutdown(
 #[no_mangle]
 pub unsafe extern "C" fn ax_resolve_sockaddr(
     node: *const c_char,
+    service: *const c_char,
     addr: *mut ctypes::sockaddr,
     len: ctypes::size_t,
 ) -> c_int {
     let name = char_ptr_to_str(node);
+    let port = char_ptr_to_str(service);
     debug!(
-        "ax_resolve_sockaddr <= {:?} {:#x} {}",
-        name, addr as usize, len
+        "ax_resolve_sockaddr <= {:?} {:?} {:#x} {}",
+        name, port, addr as usize, len
     );
     ax_call_body!(ax_resolve_sockaddr, {
-        if addr.is_null() {
+        if addr.is_null() || (node.is_null() && service.is_null()) {
             return Err(LinuxError::EFAULT);
         }
         let addr_slice = unsafe { core::slice::from_raw_parts_mut(addr, len) };
-        let res = resolve_socket_addr(name?)?;
+        let res = if let Ok(domain) = name {
+            if let Ok(a) = domain.parse::<IpAddr>() {
+                vec![a]
+            } else {
+                resolve_socket_addr(domain).map_err(|_| LinuxError::EINVAL)?
+            }
+        } else {
+            vec![Ipv4Addr::new(127, 0, 0, 1).into()]
+        };
+
         for (i, item) in res.iter().enumerate().take(len) {
-            addr_slice[i] = as_c_sockaddr(&(*item, 0).into());
+            addr_slice[i] = as_c_sockaddr(
+                &(
+                    *item,
+                    port.map(|p| p.parse::<u16>().unwrap_or(0)).unwrap_or(0),
+                )
+                    .into(),
+            );
         }
         Ok(if len > res.len() { res.len() } else { len })
     })
 }
+
+/// Query addresses for a domain name.
+///
+/// Return 0 if success.
+#[no_mangle]
+pub unsafe extern "C" fn ax_setsockopt(
+    sock_fd: c_int,
+    level: c_int,
+    optname: c_int,
+    _optval: *const c_void,
+    _optlen: ctypes::socklen_t,
+) -> c_int {
+    debug!("ax_setsockopt <= {} {} {}", sock_fd, level, optname);
+    ax_call_body!(ax_setsockopt, {
+        match level as u32 {
+            ctypes::SOL_SOCKET => match optname as u32 {
```

Comment:
Since there is no meaningful operation yet, this `ax_*` function can be removed and written in `c_libax/network.c` instead.